demos/form-control/calendar.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\ui\demo;
+
+use atk4\ui\Form;
+use atk4\ui\GridLayout;
+use atk4\ui\JsExpression;
+use atk4\ui\JsFunction;
+use atk4\ui\JsToast;
+use atk4\ui\Persistence\Type\Date;
+
+/** @var \atk4\ui\App $app */
+require_once __DIR__ . '/../init-app.php';
+
+$layout = GridLayout::addTo($app, ['rows' => 1, 'columns' => 2]);
+
+$form = Form::addTo($layout, [], ['r1c1']);
+
+Date::setFormat('date', 'Y-m-d');
+$form->addControl('date_y_m_d', [Form\Control\Calendar::class, 'type' => 'date', 'caption' => 'Date (Y-m-d)'])
```

Comment:
UI format types must be implemented in non-static way, probably once per App which we can then easily access with like `->frontFormatter` and conversion must be definitely handled in Control itself - in php, we must NEVER work on formatter data - these are only for user

Reply:
example here: for date type, type will allow to set `\DateTimeInterface` only, never formatter string from php

Reply:
Revert to previous ui_persistence for now.